docs/developer/decision-records/2022-02-07-micrometer-metrics/README.md

```diff
@@ -0,0 +1,132 @@
+# Micrometer metrics
+
+## Decision
+
+Use the [Micrometer](https://micrometer.io/) metrics library to automatically collect metrics from the host system, JVM, and frameworks and libraries used in EDC (including OkHttp, Jetty and Jersey). This enables users to integrate with most observability stacks in the market.
+
+**Note:** This decision record only deals with *automatically collected metrics*. *Custom metrics* will be addressed in further work.
+
+**Note:** The basic functionality in the EDC IoC container does not allow proper decoupling of Micrometer listener from factory logic. Therefore, it is currently necessary to write Micrometer binding code directly in factory classes (for example, the class that sets up the Jetty server).
+
+## Rationale
+
+Capturing key system metrics, especially on I/O interfaces, is essential to system observability. Micrometer is a mature framework for collecting metrics, is well supported by vendors of metrics collector products, and can directly integrate with the popular open-source libraries used in EDC.
+
+The [OpenTelemetry Metrics](https://opentelemetry.io/docs/reference/specification/metrics/) library standard was also considered, but it is currently experimental and still lacks broad vendor and library support. It is likely to catch up in the medium-term future, at which point Micrometer might be swapped.
+
+The [OpenTelemetry agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks) already supports collecting metrics from Micrometer, and [exposing them](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md) to a Prometheus endpoint. Prometheus is a widely used open source standard for exposing application metrics via an HTTP endpoint. On Azure, the [Application Insights agent](https://docs.microsoft.com/azure/azure-monitor/app/java-in-process-agent) natively forwards metrics from Micrometer to Azure Application Insights.
+
+## Spike
+
+We built a [spike](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/compare/main...agera-edc:spike/419/7-micrometer-metrics?expand=1) in a repository fork. The spike configures Micrometer to collect metrics in multiple ways.
+
+### Micrometer registry
+
+We use the [global Micrometer registry](https://micrometer.io/docs/concepts#_global_registry) to enable [Application Insights agent](https://docs.microsoft.com/azure/azure-monitor/app/java-in-process-agent) integration. The registry does not require any configuration.
+
+### Prometheus
+
+We created [a sample 04.3](https://github.com/agera-edc/DataSpaceConnector/tree/spike/419/7-micrometer-metrics/samples/04.3-open-telemetry) to deploy an open-source Prometheus server:
```

Comment:
Is this supposed to be an extension of the sample 04.3 created in #1 ? Will it be merged as well, it seems to be in a spike branch.

Reply:
Yes later, but not in this PR. Here we are just referencing the spike branch.